omod/src/main/java/org/openmrs/module/owa/web/controller/AddAppController.java

```diff
@@ -74,7 +76,7 @@ public String upload(@RequestParam("file") MultipartFile file, HttpServletReques
                     message = messageSourceService.getMessage("owa.not_a_zip");
                     log.warn("App is not a zip archive");
                     uploadedFile.delete();
-                    return message;
```

Comment:
This is also fine, but just a suggestion:
session.setAttribute(WebConstants.OPENMRS_ERROR_ATTR,"whatever your error"); 
that is used in core at a few places.